Task_1.py

```diff
@@ -1,160 +1,165 @@
-        if len(item) > 1:
-            sort = sorted(item.items(), key=itemgetter(1))
-            for i in range(1, len(sort)):
-                a = i-1
-                deltatime.append([sort[i][0], sort[a][0], sort[i][1] - sort[a][1]])
-    time_deltatime = []
+def nearest_reviews(main_dict):
+    '''function returns the shortest interval between ratings of one user and the length of both messages
+
+    Task_1.py: to create file general-stats.cvs containing information about
+    the shortest interval between ratings of one user (among all users) and
+    the length of both messages which create this interval;
+    '''
+
+    def dict_timereviews_of_user(main_dict):
+        '''function returns the dictionary following view
+        all_timereview = {
+                            reviewerName: {IDreview: unixReviewTime,
+                                            IDreview: unixReviewTime,
+                                                ...},
+                            reviewerName: {IDreview: unixReviewTime,
+                                            IDreview: unixReviewTime,
+                                                ...}
+                        ...}'''
+        all_timereview = {}
+        for key, value in main_dict.items():
+            if 'reviewerName' in value.keys() and 'unixReviewTime' in value.keys() \
+                    and value['reviewerName'] != 'Amazon Customer':
+                try:
+                    all_timereview[value['reviewerName']].setdefault(key, value['unixReviewTime'])
+                except:
+                    all_timereview.setdefault(value['reviewerName'], {key: value['unixReviewTime']})
+        return all_timereview
+
+    def deltatime(all_timereview=dict_timereviews_of_user(main_dict)):
+        '''function returns the time between nearest user's reviews like following list 'deltatime':
+            [[ IDreview_1, IDreview_2, time_between_reviews], [IDreview_n, IDreview_m, time_between_reviews]]'''
+
+        # argument sorted_all_timereview returns the lists consists of all reviews of every user,
+        # who sent more than one review, in ascending order
+        # [[(IDreview: unixReviewTime), (IDreview: unixReviewTime), ..., (IDreview: unixReviewTime)], [...],
+        # [...], [(IDreview: unixReviewTime), (IDreview: unixReviewTime), ..., (IDreview: unixReviewTime)]]
+        sorted_all_timereview = [sorted(item.items(), key=itemgetter(1))
+                                 for item in all_timereview.values() if len(item) > 1]
+
+        # 'deltatime'=[[IDreview_1, IDreview_2, time_between_reviews], [IDreview_n, IDreview_m, time_between_reviews]]
+        deltatime = [[item[int(i-1)][0], item[i][0], item[i][1] - item[int(i-1)][1]]
+                     for item in sorted_all_timereview for i in range(1, len(item))]
+        return deltatime
+
+    nearest_reviews = min(
+                    list(filter(lambda comment: comment[2] != 0, deltatime())),
```

Comment:
`list(filter(lambda comment: comment[2] != 0, deltatime()))` called twice, it's not very efficient